trainer/spec/test_parser_spec.rb

```diff
@@ -26,6 +26,35 @@
       end
     end
 
+    describe "#generate_cmd_parse_xcresult" do
+      let(:xcresult_sample_path) { "./trainer/spec/fixtures/Test.test_result.xcresult" }
+      let!(:subject) { Trainer::TestParser.new(xcresult_sample_path) }
+      let(:command) { subject.send(:generate_cmd_parse_xcresult, xcresult_sample_path) }
+
+      before do
+        allow(File).to receive(:expand_path).with(xcresult_sample_path).and_return(xcresult_sample_path)
+        allow_any_instance_of(Trainer::TestParser).to receive(:`).with('xcrun xcresulttool version').and_return(version)
+      end
+
+      context 'with >= Xcode 16 beta 3' do
+        let(:version) { 'xcresulttool version 23021, format version 3.53 (current)' }
+        let(:expected) { "xcrun xcresulttool get --format json --path #{xcresult_sample_path} --legacy" }
+
+        it 'should pass `--legacy`', requires_xcode: true do
```

Comment:
What is `requires_xcode`?

Reply:
OK, it's better to add this tag.

Reply:
In this case, we only check the command builder. It doesn't depend on Xcode, so we can get rid of this.

Reply:
if not adding `requires_xcode` flag, ubuntu test will fail on CI.

https://app.circleci.com/pipelines/github/fastlane/fastlane/6614/workflows/3db274a0-fb41-4ce4-9783-a50fda4ff9c2/jobs/101945?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link&utm_content=summary

```
Failures:

  1) Trainer Trainer::TestParser#generate_cmd_parse_xcresult with >= Xcode 16 beta 3 should pass `--legacy`
     Failure/Error: let!(:subject) { Trainer::TestParser.new(xcresult_sample_path) }

     Errno::ENOENT:
       No such file or directory - xcrun
```